react/src/users/UsersContainer.js

```diff
@@ -1,22 +1,62 @@
 import { withRouter } from 'react-router-dom';
 import axios from 'axios';
-import CircularProgress from '@mui/material/CircularProgress';
+import { CircularProgress, Alert } from '@mui/material';
 import React, { useState, useEffect } from 'react';
 import ROUTES_USERS from './routes';
 import UsersPage from './components/UsersPage';
 
-const UsersContainer = () => {
+const UsersContainer = (props) => {
+  const { currentUser } = props;
   const [loading, setLoading] = useState(false);
   const [users, setUsers] = useState([]);
-  const [userUpdates, setUserUpdates] = useState([]);
-
+  const [newUser, setNewUser] = useState('');
+  const [userUpdates, setUserUpdates] = useState({});
```

Comment:
Instead of using the `userUpdates` state, another way to do this would be to modify the `users` state directly since that already includes the permissions. Of course, you will probably find it difficult to maintain the immutability of the state given the nesting; to address this, I would recommend Immer (https://immerjs.github.io/immer/)

Then, upon the user clicking 'save', we could send the `users` state to the backend and, within an atomic transaction, delete every `UserPermission` and bulk create new `UserPermission` records. At least that's make makes sense to me given the UI + data structure...